autotest/pyscripts/test_gdal2tiles.py

```diff
@@ -101,10 +101,38 @@ def test_gdal2tiles_py_cleanup():
 
     return 'success'
 
+
+def test_does_not_error_when_source_bounds_close_to_tiles_bound():
+    """
+    Case where the geo coords of the image top Y point gives a corresponding tile of X, Y, Z
+    BUT where only a portion of the pixels of the first pixel line would be on this tile.
+    When computing the raster for the tile, we don't get any data (the pixel line is "assigned" to
+    the tile below)
+    """
+    out_folder = 'tmp/out_gdal2tiles_bounds_approx'
+    try:
+        shutil.rmtree(out_folder)
+    except:
+        pass
+
+    script_path = test_py_scripts.get_py_script('gdal2tiles')
+    if script_path is None:
+        return 'skip'
+
+    # Will blow an exception if the case is not handled properly
+    test_py_scripts.run_py_script(
```

Comment:
ok so you are right. This code is in `run_test` but I'm actually inside and have an opportunity to catch it.
But I found why it did not work. `run_py_script_as_py_module` is redirection stdout while it executes the script and if the execution fails with an error, the standard stdout is never put back! Thus the `post_failure` would not work....
I'll look into fixing that.